File: csharp/Algorithms.Tests/BinarySearchTest.cs
using System;
using Xunit;
using Algorithms;

namespace Algorithms.Tests
{
    public class BinarySearchTest
    {
        [Fact]
        public void Test()
        {
            int[] myList = {1,3,5,7,9};
            Assert.True(BinarySearch.Search(myList,3) == 1);
            Assert.Null(BinarySearch.Search(myList,-1));
        }
    }
}

File: java/src/test/java/grokking/algorithms/BinarySearchTest.java
package grokking.algorithms;

import org.junit.Test;

/**
 * Unit test for simple App.
 */
public class BinarySearchTest 
{
    /**
     * Rigorous Test :-)
     */
    @Test
    public void test()
    {
        int[] myList = {1,3,5,7};
        org.junit.Assert.assertTrue(BinarySearch.search(myList, 3) == 1);
        org.junit.Assert.assertNull(BinarySearch.search(myList, -1));
    }
}
